.github/start-wasm-example/tests/wasm_example.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { test, expect, Page } from '@playwright/test';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('http://localhost:8000/');
+});
+
+test.describe('WASM example', () => {
+  test('Wait for success', async ({ page }, test_info) => {
+    let start = new Date().getTime();
+
+    let found = false;
+    while (new Date().getTime() - start < 600000) {
+      let msg = await promise_with_timeout(100, on_console(page), "no log found");
+      if (msg.includes("no log found")) {
+        continue;
+      }
+      console.log(msg);
+      if (msg.includes("Test successful")) {
+        await page.screenshot({ path: `screenshot-${test_info.project.name}.png`, fullPage: true });
+        found = true;
+        break;
+      }
+    }
+
+    expect(found).toBe(true);
+  });
+
+});
+
+function on_console(page) {
+  return new Promise(resolve => {
+    page.on('console', msg => resolve(msg.text()));
+  });
+}
+
+async function promise_with_timeout(time_limit, task, failure_value) {
+  let timeout;
+  const timeout_promise = new Promise((resolve, reject) => {
+    timeout = setTimeout(() => {
+      resolve(failure_value);
+    }, time_limit);
+  });
+  const response = await Promise.race([task, timeout_promise]);
+  if (timeout) {
+    clearTimeout(timeout);
```

Comment:
I don't think this exception safe cleaning up

I'm not sure I really care though.

Reply:
I *think* it should be ok 🤷

Reply:
This is why we use Rust, because then we don't have to think about such *insert choice word here*

But yeah, I agree that we can just ignore this.